src/utils/check-connection.ts

```diff
@@ -1,33 +1,44 @@
-import { Socket } from "net";
+import * as retry from "retry";
 
-export const throwIfCannotConnectToOrigin = async (url: string) => {
-  const { hostname, port, protocol, origin } = new URL(url);
-  const defaultPortForProtocol = protocol === "https:" ? 443 : 80;
-  const portNumber =
-    port != null && port != "" ? Number(port) : defaultPortForProtocol;
-  const connectionAccepted = await canConnectTo(hostname, portNumber);
-  if (!connectionAccepted) {
-    throw new Error(
-      `Could not connect to '${hostname}:${portNumber}'. Please check:\n\n` +
-        `1. The server running at '${origin}' has fully started by the time the Meticulous action starts. You may need to add a 'sleep 30' after starting the server to ensure that this is the case.\n` +
-        `2. The server running at '${origin}' is using tcp instead of tcp6. You can use 'netstat -tulpen' to see what addresses and ports it is bound to.\n\n`
+export const throwIfCannotConnectToOrigin = async (appUrl: string) => {
+  // Wait 10s, 20s, 40s, 80s for a total of 2.5 minutes
+  const operation = retry.operation({
+    retries: 4,
+    factor: 2,
+    minTimeout: 10_000,
+  });
```

Comment:
[nit] Maybe?:

```suggestion
  // Wait 1s, 2s, 4s, 8s, 16s, 32s, 64s for a total of just over 2 minutes
  const operation = retry.operation({
    retries: 7,
    factor: 2,
    minTimeout: 1_000,
  });
```

Since the op should be pretty fast, and this should mean that if the app just takes an extra few seconds to start up then the tests will be a bit faster.

Reply:
Ah "Seting this to 1 means do it once, then retry it once."

Reply:
Oh I guess we do have a 5s timeout on the op. Not sure what happens when these things queue up.

Reply:
Yeah, I didn't want to risk any weird race conditions, nor reduce the current timeout of 5 seconds.

Reply:
I'm guessing that the time taken for attempt to resolve does not affect the time between retries, so in the case of 5s timeouts it'll do: 0s, 6s, 13s etc. (rather than 0s, 1s, 2s etc. with attempts running in parrelel, or 0s, 5s, 10s, 15s, 20s, 25s, 32s, 64s). In which case the shorter retries make sense since most of the time it'll probably just return a 502 in milliseconds.